src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2327,6 +2331,229 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
+
+                    // Get children properties with Name == name
+                    if (parentProperty.EnumChildren(enum_DEBUGPROP_INFO_FLAGS.DEBUGPROP_INFO_PROP, 0, Guid.Empty, 0, name, 0, out IEnumDebugPropertyInfo2 children) == HRConstants.S_OK)
+                    {
+                        if (children.GetCount(out uint childrenCount) == HRConstants.S_OK && childrenCount == 1)
+                        {
+                            DEBUG_PROPERTY_INFO[] propertyInfo = new DEBUG_PROPERTY_INFO[1];
+                            if (children.Next(childrenCount, propertyInfo, out _) == HRConstants.S_OK)
+                            {
+                                property = propertyInfo[0].pProperty;
+                            }
+                        }
+                        else
+                        {
+                            errorMessage = $"Unable to get property from parent: there are {childrenCount} children properties with name {name}";
+                        }
+                    }
+                    else
+                    {
+                        errorMessage = "ParentProperty.EnumChildren failed";
+                    }
+                }
+            }
+
+            if (errorMessage != null)
+            {
+                response.Description = errorMessage;
+                responder.SetResponse(response);
+                return;
+            }
+
+            // If we don't have a property for what we're trying to set the data bp on yet, attempt to get it by evaluating the variable name.
+            if (property == null)
+            {
+                // If we don't have a frame, just use top frame.
+                if (frame == null)
+                {
+                    m_frameHandles.TryGetFirst(out frame);
+                }
+                TryEvaluate(responder.Arguments.Name, EvaluateArguments.ContextValue.Unknown, frame, out _, out property, out errorMessage);
+            }
+
+            if (errorMessage != null)
+            {
+                response.Description = errorMessage;
+                responder.SetResponse(response);
+                return;
+            }
+
+            if (property != null && property is IDebugProperty160 property160)
+            {
+                // If we have a property that we can get the address/size from
+                if (property160.GetDataBreakpointInfo160(out string address, out uint size, out string displayName, out errorMessage) == HRConstants.S_OK)
+                {
+                    // Test to see if we can set a data bp
+                    AD7BreakPointRequest pBPRequest = new AD7BreakPointRequest(address, size);
+                    int hr = m_engine.CreatePendingBreakpoint(pBPRequest, out IDebugPendingBreakpoint2 pendingBp);
+                    if (hr == HRConstants.S_OK && pendingBp != null)
+                    {
+                        hr = pendingBp.Bind();
+                    }
+                    pendingBp?.Delete();
```

Comment:
Remove. You don't want to bind breakpoints as port of DataBreakpointInfoRequest. If we need any more error checking it should be done in GetDataBreakpointInfo160.